### PR TITLE
Fix the issue that text and bg color in PopupEntry is not changed while edit text

### LINKED ISSUE #233: PopupEntry, inability to see changes while Text edit view is active.
**Describe the bug**

PopupEntry properties such as TextColor and PopupBackgroundColor are not changeable while the popup entry is visible.  Rather than an enhancement, i believe this is a bug.

Given the code below, i have a popup entry for a  user to enter a valid hostname.
While the user is entering text the input is validated.
If the text is valid, the text should be made green, if the text is invalid the text should be made red, but as can be seen, the color change is not applied until the keyboard/editor is closed.

It should be possible to verify and visually adjust the TextColor or PopupBackgroundColor color while a user is actively inputting text, but it is not currently possible to see these changes in the edit view, only after the editor/keyboard is closed.  This may appear to me a bug rather than an enhancement.

Note: this bug is only observed on emulator as i do not (yet) own real hardware.

```
private void OnListCellTap(Credentials credentials)
{
	var hostEntry = new PopupEntry
	{
		Margin = new Thickness(20, 0, 20, 0),
		HeightRequest = 65,
		HorizontalOptions = LayoutOptions.Fill,
		VerticalOptions = LayoutOptions.Fill,
		HorizontalTextAlignment = TextAlignment.Center,
		BackgroundColor = Color.Black,
		PopupBackgroundColor = Color.Black,
		ReturnType = ReturnType.Done,
	};
	hostEntry.TextChanged += (sender, args) =>
	{
		if (ValidHostname.IsMatch(hostEntry.Text))
		{
			hostEntry.TextColor = Color.Green;
			hostEntry.BackgroundColor = Color.Black;
		}
		else
			hostEntry.TextColor = Color.Red;

		hostEntry.BackgroundColor = string.IsNullOrWhiteSpace(hostEntry.Text) ? Color.Red : Color.Black;
	};

	hostEntry.Text = credentials.Name;
			
	_profileEditView = new StackLayout
	{
		Padding = new Thickness(20, 30, 20, 20),
		Margin = new Thickness(20, 20, 20, 20),
		Children = { hostEntry }
	};

	Content = _profileEditView;
}
```

**To Reproduce**
1. start a edit text with a valid text, here we see the color green
image

2. Then edit the test with an invalid character, on text apply a text color change.
3. Observe the text change does not apply in the edit view.
image

4. Immediatly close the editor and then see the color change applied.
image

Note that the same is true for PopupBackgroundColor, changes to it while editor/keyboard active are not observed until the editor/keyboard is closed and then re opened, so observing the changes is not only 1, but 2 steps away.

**Expected behavior**
It is expected that changes to TextColor and PopupBackgroundColor are applied during edit view as they are direct properites of that view.

**Screenshots**
see above

**Environment (please complete the following information):**
 - Target Product:
image

 - Tizen Platform Version:
 4.0.0
 - Tizen.CircularUI Version:
1.4.0

**Additional context**
On a related note:
Android specifically separates the Hint, Displayed and Changing text properties of an EditText object.
While i dont see the pressing need to separate the displayed and changing text, adding a PopupEntry object Hint property seems like a nice enhancement.

### PR DESCRIPTION
**Pull request by jkpu**
### Description of Change ###
Fix the issue that text and bg color in PopupEntry is not changed while edit text

### Bugs Fixed ###
- Fix #233

### API Changes ###
None

### Behavioral Changes ###
None

### CODE REVIEW

File: src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs [hunk 1]
```diff
@@ -212,6 +213,11 @@ void UpdatePopupBackgroundColor()
             else
             {
                 _popupBackgroundColor = bgColor.ToNative();
+
+                if(_IsPopupOpened == true && _editorPopup != null)
+                {
+                    _editorPopup.Color = _popupBackgroundColor;
+                }
```
Comment (myroot) [inline src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs]: It does not handle a Color.Default case, 

``` c#
        void UpdatePopupBackgroundColor()
        {
            Xamarin.Forms.Color bgColor = ((PopupEntry)Element).PopupBackgroundColor;
            if (bgColor == Xamarin.Forms.Color.Default)
            {
                _popupBackgroundColor = DefaultColor;
            }
            else
            {
                _popupBackgroundColor = bgColor.ToNative();
            }

            if (_IsPopupOpened)
            {
                _editorPopup.Color = _popupBackgroundColor;
            }
        }
```

File: src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs [hunk 2]
```diff
@@ -248,6 +254,17 @@ void UpdateIsPopupOpened()
                 HidePopup();
             }
         }
+
+
+        protected override void UpdateTextColor()
```
Comment (myroot) [inline src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs]: Before, We do not apply TextColor on PopupEntry, If you should apply TextColor on PopupEntry, we also change `CreatePopup` method to change _editor.TextColor

Comment (jkpu) [reply src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs]: TextColor set in ShowPopup().

            _editor.TextStyle = Control.TextStyle;
            _editorPopup.Color = _popupBackgroundColor;

File: src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs [hunk 3]
```diff
@@ -248,6 +254,17 @@ void UpdateIsPopupOpened()
...
+        {
+            base.UpdateTextColor();
+
+            if (_IsPopupOpened && _editor != null)
+            {
+                _editor.TextColor = Element.TextColor.ToNative();
```
Comment (myroot) [inline src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs]:
```suggestion
                _editor.TextStyle= Control.TextStyle;
```

File: src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs [hunk 4]
```diff
@@ -64,6 +66,30 @@ protected override void OnElementChanged(ElementChangedEventArgs<Xamarin.Forms.E
...
         }
 
+        protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            base.OnElementPropertyChanged(sender, e);
+            if (_IsPopupOpened && _editor != null)
+            {
+                if (e.PropertyName == XEntry.TextColorProperty.PropertyName)
+                {
+                    _editor.TextColor = Element.TextColor.ToNative();
```
Comment (myroot) [inline src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs]: How about use Control?
```suggestion
                    _editor.TextColor = Control.TextColor;
```
or TextStyle?
``` c#
    if (e.PropertyName == XEntry.TextColorProperty.PropertyName ||
        e.PropertyName == XEntry.FontSizeProperty.PropertyName ||
        e.PropertyName == XEntry.FontFamilyProperty.PropertyName ||
        e.PropertyName == XEntry.FontAttributesProperty.PropertyName)
    { 
       _editor.TextStyle = Control.TextStyle;
    }
```

Comment (jkpu) [reply src/Tizen.Wearable.CircularUI.Forms.Renderer/PopupEntryRenderer.cs]: Done.  set  Control.TextStyle to popup editor's TextStyle.

**Pull request merged**